### PR TITLE
Show yanked reason on release page

### LINKED ISSUE #9837: Public release history and release page do not show yanked reason
**What's the problem this feature will solve?**
For yanked versions with a reason provided, PyPI shows this reason on neither the public release history nor the public release page.

**Describe the solution you'd like**
I don't know where exactly it should be shown but I think it should be shown somewhere *at least* on the release page.

**Additional context**
See version 3.4.8 that *has* a yanked reason but it is not shown on the page:
https://pypi.org/project/Red-DiscordBot/#history
https://pypi.org/project/Red-DiscordBot/3.4.8/

### PR DESCRIPTION
**Pull request by cofiem**
Releases that have been yanked can include a reason. This reason is not currently displayed on the PyPI website.

For releases that have been yanked, show the yanked reason in a callout box on the project's release page.

Closes #9837

See screenshot for an example.
image

### CODE REVIEW

File: warehouse/templates/packaging/detail.html [hunk 1]
```diff
@@ -282,6 +282,12 @@ <h3 class="sidebar-section__title">{% trans %}Navigation{% endtrans %}</h3>
 
         {# Tab: Project description #}
         <div id="description" data-project-tabs-target="content" class="vertical-tabs__content" role="tabpanel" aria-labelledby="description-tab mobile-description-tab" tabindex="-1">
+          {% if release.yanked and release.yanked_reason %}
+            <div class="callout-block callout-block--warning">
```
Comment (miketheman) [inline warehouse/templates/packaging/detail.html:286]: Nice usage of the blocks - had you considered using the `callout-block--danger` instead ([ref](https://github.com/pypi/warehouse/blob/695a3b9549cb4cdf8226aa2d57ab7d419af25c84/warehouse/static/sass/blocks/_callout-block.scss#L36-L37))? That lines up more clearly with the badge at the top of the page.
Screenshot 2023-12-18 at 10 51 17

**Review by miketheman — COMMENTED** [review]
Generally good - had a question about the color choice.

File: warehouse/templates/packaging/detail.html (hunk 1 above)
Comment (cofiem) [reply warehouse/templates/packaging/detail.html:286]: I have no personal opinion on either the warning or danger callout, so I'll go with the danger one.

**Review by miketheman — APPROVED** [review]
Excellent, looks great! Thanks!

**Comment by miketheman** [comment]
Keep in mind that due to caching, this may not appear on existing yanked releases until their cache expires.
Any new yanked releases automatically purge their cached page.

**Comment by di** [comment]
We might want to think about finding a way to render links when they are present in the yank reason, e.g. https://pypi.org/project/pip/21.2/

**Pull request merged**